File: perl/dataprocess/cal_yearly.pl
#!/usr/bin/perl -w

# Description: Calculate yearly from monthly data of PCOM
#
#       Usage: ./xxx.pl
#
#      Author: OU Yuyuan <ouyuyuan@lasg.iap.ac.cn>
#     Created: 2013-09-16 21:32:38 CST
# Last Change: 2013-09-17 18:48:50 CST

use strict;
use 5.010;

my $year = 1;
my $total_year = 100;
my $src_dir = "monthly";
my $des_dir = "yearly";

while ($year <= $total_year) {
   my $month = 1;
   my $total_mon;
   my $filename;
   my $cmd = "cdo ensmean ";

   while ($month <= 12) {
      $total_mon = ($year-1)*12 + $month;
      my $filename = sprintf("%s%08s%s","$src_dir/N",$total_mon,".nc");
#      say $filename;

      $cmd .= $filename . " ";
      $month += 1;
   }
   $cmd .= sprintf("%s%04s%s","$des_dir/year_",$year,".nc");

   say $cmd;
   system($cmd);

   $year += 1;
}

File: calculate/ou101/scripts/pcom/cal_yearly.py
#!/usr/bin/env python

# Description: cal. climate mean of PCOM monthly output data
#
#      Author: OU Yuyuan <ouyuyuan@lasg.iap.ac.cn>
#     Created: 2013-12-12 10:02:20 BJT
# Last Change: 2014-12-12 19:22:51 BJT

import os
import sys

#datDir = "/snfs01/ou/models/pcom_1.0/exp/exp24_taux_an_40y_N/output"
datDir = "/snfs01/ou/models/pcom_1.0/exp/exp_spinup"
#datDir = "/snfs01/ou/models/pcom_1.0/exp/exp17_ws_control_cycle_re"
yr_sta = 1
yr_end = 15

os.chdir(datDir)
for year in range(yr_sta, yr_end + 1):
  mn_sta = (year - 1) * 12 + 1
  mn_end = year * 12
  filenames = ''
  for i in range(mn_sta, mn_end + 1):
    f = "monthly/N%08d.nc " % i
    filenames += f
  cmd = 'cdo ensmean ' + filenames +  ' yearly/year_%04d.nc' % year
  print(cmd)
  os.system(cmd)
